src/main/java/org/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -79,15 +91,24 @@
 
     private final EntryEditorPreferences preferences;
     private final DialogService dialogService;
+    private final NewDroppedFileHandler fileHandler;
 
-    public EntryEditor(BasePanel panel, EntryEditorPreferences preferences, FileUpdateMonitor fileMonitor, DialogService dialogService) {
+    public EntryEditor(BasePanel panel, EntryEditorPreferences preferences, FileUpdateMonitor fileMonitor, DialogService dialogService, ExternalFileTypes externalFileTypes) {
         this.panel = panel;
         this.databaseContext = panel.getBibDatabaseContext();
         this.undoManager = panel.getUndoManager();
         this.preferences = Objects.requireNonNull(preferences);
         this.fileMonitor = fileMonitor;
         this.dialogService = dialogService;
 
+        fileHandler = new NewDroppedFileHandler(dialogService, databaseContext, externalFileTypes,
+                                                Globals.prefs.getFileDirectoryPreferences(),
```

Comment:
please remove all these globals...(e.g. add the prefs needed to `EntryEditorPreferences`).